torch/_dynamo/source.py

```diff
@@ -399,10 +399,15 @@ def __post_init__(self):
         assert self.base is not None
 
     def reconstruct(self, codegen):
```

Comment:
This is nearly identical to the standard `GetItemSource` above, can you abstract this out to a common place?

Reply:
As far as I understand, the identical parts of the 2 are:
- Reconstruct the `base`
- Check if the `index` is an instance of `Source`
- Maybe (check the differences) load the `index` if it is not 

The differences are:
- `GetItemSource` still has to check if `index` is a slice
- `GetItemSource` creates a `BINARY_SUBSCR` instruction
- `ODictGetItemSource` has to load the `getitem` function
- `ODictGetItemSource` has to call this function (another bytecode)

Do you think that I should do it, regardless of the differences?

Reply:
If you abstract out a bytecode_suffix method that is overridden in each subclass, and say index_is_slice is always false for OdictGetItem, I think the differences resolve, right?

Reply:
Yes. You are totally right. Missed that completely.